### PR TITLE
AP_Compass: Add in QMC5883P driver

### PR DESCRIPTION
**Pull request by Jono453**
As discussed previously in https://github.com/ArduPilot/ardupilot/pull/19841, we would like to keep the QMC533P and QQMC533L seperate. The QMC533P build was tested on an AP_Periph build on a F405 and was working with valid readings from the onboard sensor.

### CODE REVIEW

**Comment by tridge** [comment]
@peterbarker I'd like to discuss approaches to new compass drivers that won't cause us to overflow flash on low flash boards but still allow flexibility to enable on custom.ardupilot.org. The problem with PRs like this one is it takes up flash on 1M boards for a sensor only used by one vendor
some options:
 - we could only problem for these uncommon new sensors with hwdef.dat line, just by removing the code in Compass::_probe_external_i2c_compasses()
 - we could add each new compass in build_options.py, and default off or default off on 1M boards

Note that this PR as it stands now adds an hwdef.dat line for this compass which never actually gets used as the compass instead gets found via Compass::_probe_external_i2c_compasses()

**Comment by Jono453** [comment]
> @peterbarker I'd like to discuss approaches to new compass drivers that won't cause us to overflow flash on low flash boards but still allow flexibility to enable on custom.ardupilot.org. The problem with PRs like this one is it takes up flash on 1M boards for a sensor only used by one vendor some options:
> 
> * we could only problem for these uncommon new sensors with hwdef.dat line, just by removing the code in Compass::_probe_external_i2c_compasses()
> * we could add each new compass in build_options.py, and default off or default off on 1M boards
> 
> Note that this PR as it stands now adds an hwdef.dat line for this compass which never actually gets used as the compass instead gets found via Compass::_probe_external_i2c_compasses()

@tridge @peterbarker what was the outcome of the discussion regarding adding the new compass driver to Ardupilot?

File: libraries/AP_Compass/AP_Compass_config.h [hunk 1]
```diff
@@ -112,6 +112,10 @@
 #define AP_COMPASS_MMC5XX3_ENABLED AP_COMPASS_I2C_BACKEND_DEFAULT_ENABLED
 #endif
 
+#ifndef AP_COMPASS_QMC5883P_ENABLED
+#define AP_COMPASS_QMC5883P_ENABLED AP_COMPASS_I2C_BACKEND_DEFAULT_ENABLED
```
Comment (tridge) [inline libraries/AP_Compass/AP_Compass_config.h:116]: to reduce impact on low flash boards I think we need && BOARD_FLASH_SIZE > 1024

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Compass/AP_Compass_config.h (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_Compass/AP_Compass_config.h:116]: We should probably get into the habit of putting these into the various minimize includes

Comment (Jono453) [reply libraries/AP_Compass/AP_Compass_config.h:116]: Done this change

**Review by tridge — APPROVED** [review]

**Pull request merged**